llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py

```diff
@@ -463,31 +463,38 @@ def get_rel_map(
         triples = []
 
         ids = [node.id for node in graph_nodes]
-        # Needs some optimization
-        response = self.structured_query(
-            f"""
-            MATCH (e:`__Entity__`)
-            WHERE e.id in $ids
-            MATCH p=(e)-[r*1..{depth}]-(other)
-            WHERE ALL(rel in relationships(p) WHERE type(rel) <> 'MENTIONS')
-            UNWIND relationships(p) AS rel
-            WITH distinct rel
-            WITH startNode(rel) AS source,
-                type(rel) AS type,
-                endNode(rel) AS endNode
-            RETURN source.id AS source_id, [l in labels(source) WHERE l <> '__Entity__' | l][0] AS source_type,
-                    source{{.* , embedding: Null, id: Null}} AS source_properties,
-                    type,
-                    endNode.id AS target_id, [l in labels(endNode) WHERE l <> '__Entity__' | l][0] AS target_type,
-                    endNode{{.* , embedding: Null, id: Null}} AS target_properties
-            LIMIT toInteger($limit)
-            """,
-            param_map={"ids": ids, "limit": limit},
-        )
-        response = response if response else []
+
+        responses = []
+        for id in ids:
+            if limit <= 0:
+                break
+            # Needs some optimization
+            response = self.structured_query(
```

Comment:
Now we are making a db query per id -- this is extremely inefficient

Can't we use the "power of cypher" to force some ordering of the nodes to match the ordering of the IDs?

Reply:
Note: I am not a cypher expert, but it seems like a single query should be possible

Reply:
I've modified the code to only make one call to the DB and force ordering of the IDs.

Reply:
Please review the new query